openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIRequestBodyMapper.java

```diff
@@ -228,31 +243,40 @@ private void handleMultipleMIMETypes(RequestBody bodyParameter,
                                          RequiredParameterNode payloadNode, Map<String, Schema> schema) {
 
         for (MappingFieldNode fieldNode: fields) {
-            if (fieldNode.children() != null) {
-                ChildNodeList nodeList = fieldNode.children();
-                for (Node nextNode : nodeList) {
-                    if (nextNode instanceof ListConstructorExpressionNode) {
-                        SeparatedNodeList mimeList = ((ListConstructorExpressionNode) nextNode).expressions();
+            if (fieldNode instanceof SpecificFieldNode) {
+                if (((SpecificFieldNode) fieldNode).fieldName().toString().equals(MEDIA_TYPE)) {
+                    if (fieldNode.children() == null) {
+                        continue;
+                    }
```

Comment:
```suggestion
            if (!(fieldNode instanceof SpecificFieldNode) || !((SpecificFieldNode) fieldNode).fieldName().toString().equals(MEDIA_TYPE) || fieldNode.children() == null) {
                  continue;
              }
```

Reply:
> if (!((SpecificFieldNode) fieldNode).fieldName().toString().equals(MEDIA_TYPE) ||
                        fieldNode.children() == null) {
                    continue;
                }
narrow down with the last two conditions since to have cast the field by passing the first condition 🙂